addon/components/modal.js

```diff
@@ -193,6 +193,28 @@ export default Component.extend({
 	 */
 	_remove() {
 		this.get('modal.content').removeObject(this.get('model'));
-	}
+	},
 
+	actions: {
+
+		/**
+		 * Action to resolve the underlying modal promise directly from the,
+		 * template, using the passed arguments as resolution values
+		 *
+		 * @method resolveModal
+		 */
+		resolveModal() {
```

Comment:
`resolve`/`reject`?? I know that I'm writing code in a modal component so the name is redundant.